From 02ba2f361e1ae64e6e27772c08a142c3f8326503 Mon Sep 17 00:00:00 2001
From: Robert Hodaszi <robert.hodaszi@digi.com>
Date: Thu, 17 Feb 2022 11:47:12 +0100
Subject: [PATCH] rc.apparmor.functions: don't use non-existing tools

DAL doesn't have 'getconf', and xargs doesn't know '-d' and
'--max-procs' arguments. Removing these. The drawback is that
rule parsing is not longer parallelized when there's a broken rule.
---
 parser/rc.apparmor.functions | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/parser/rc.apparmor.functions b/parser/rc.apparmor.functions
index 8dec7f6..972de4a 100644
--- a/parser/rc.apparmor.functions
+++ b/parser/rc.apparmor.functions
@@ -183,8 +183,7 @@ __parse_profiles_dir() {
 			fi
 			echo "$profile"
 		done | \
-		# Use xargs to parallelize calls to the parser over all CPUs
-		xargs -n1 -d"\n" --max-procs="$(getconf _NPROCESSORS_ONLN)" \
+		xargs -n1 \
 			"$PARSER" $PARSER_OPTS "$parser_cmd" --
 		if [ $? -ne 0 ]; then
 			status=1
-- 
2.27.0

